time-manager/System/ThreadManager.hs

```diff
@@ -209,7 +210,7 @@ labelMe l = do
     labelThread tid l
 
 withHandle
-    :: ThreadManager -> T.TimeoutAction -> (T.Handle -> IO a) -> IO (Maybe a)
```

Comment:
Maybe also add this change (and the one in `System.TimeManager`) to the `ChangeLog.md`? It technically is a breaking change (which is fine since we're bumping a major version)

Reply:
> Yes, but the throwing of TimeoutThread is expected and caught with handle, so it doesn't actually kill the current thread, it kills the action and continues with the current thread. Right?

There are only tow threads: TimerManager (a) and the thread to call `withHandleKillThread` (b).
(a) throws `TimeoutThread` to (b).
(b)'s `ignore` catches it.
At this moment, (b)'s `action` has been terminated already.
`ignore` does not re-throw the exception and finished.
So, (b) dies.